test/tools/d_do_test.d

```diff
@@ -532,20 +529,29 @@ int main(string[] args)
 
 int tryMain(string[] args)
 {
-    if (args.length != 4)
+    if (args.length != 2)
     {
         usage();
         return 1;
     }
 
-    string input_dir      = args[1];
-    string test_name      = args[2];
-    string test_extension = args[3];
+    auto test_file = args[1];
+    string input_dir = test_file.dirName;
+    string test_base_name = test_file.baseName;
+    string test_name = test_base_name.stripExtension();
```

Comment:
This is a bit inconsistent here. All three are functions ;-)
(by "your majority vote" maybe adapt stripExtension?)